interp/constrextern.ml

```diff
@@ -307,8 +307,25 @@ let extern_evar n l = CEvar (n,l)
     For instance, in the debugger the tables of global references
     may be inaccurate *)
 
+let rec dirpath_of_modpath = function
+  | MPfile dp -> dp
+  | MPbound mbid -> let (_,id,_) = MBId.repr mbid in DirPath.make [id]
+  | MPdot (t, l) -> Libnames.add_dirpath_suffix (dirpath_of_modpath t) (Label.to_id l)
+
+let path_of_global = function
+  | GlobRef.VarRef id -> Libnames.make_path DirPath.empty id
+  (* We rely on the tacite invariant that the label of a constant is used to build its internal name *)
+  | GlobRef.ConstRef cst -> Libnames.make_path (dirpath_of_modpath (Constant.modpath cst)) (Label.to_id (Constant.label cst))
+  (* We rely on the tacite invariant that an inductive block inherits the name of its first type *)
+  | GlobRef.IndRef (ind,1) -> Libnames.make_path (dirpath_of_modpath (MutInd.modpath ind)) (Label.to_id (MutInd.label ind))
+  (* These are hacks *)
+  | GlobRef.IndRef (ind,n) -> Libnames.make_path (dirpath_of_modpath (MutInd.modpath ind)) (Id.of_string ("unknown_" ^ String.ordinal n ^ "_type_defined_mutually_with_" ^ Label.to_string (MutInd.label ind)))
+  | GlobRef.ConstructRef ((ind,1),p) -> Libnames.make_path (dirpath_of_modpath (MutInd.modpath ind)) (Id.of_string ("unknown_" ^ String.ordinal (p+1) ^ "_constructor_of_" ^ Label.to_string (MutInd.label ind)))
+  | GlobRef.ConstructRef ((ind,n),p) -> Libnames.make_path (dirpath_of_modpath (MutInd.modpath ind)) (Id.of_string ("unknown_" ^ String.ordinal (p+1) ^ "_constructor_of_" ^ String.ordinal n ^ "_type_defined_mutually_with_" ^ Label.to_string (MutInd.label ind)))
```

Comment:
I don't think that such a verbose display you've just introduced is very useful, it's drowning the information in an ocean of irrelevance and is going to clutter the output. Simply following a scheme like `<inductive-foo:1:2>` or something should be enough, as long as it is properly documented. In particular the use of `ordinal` is completely overblown.

Reply:
That's a bit too verbose, I agree. I removed the `<`/`>` and now used `inductive_foo_1`, `constructor_foo_1`, and `constructor_foo_1_2`.

Reply:
You should still use chevrons in order to guarantee that this will not clash with existing constructors and statically ensure this will never be parsable as an identifier.

Reply:
Ok, I rally. Then using `Id.of_string_soft`.